k8s/gvclient.go

```diff
@@ -617,29 +617,28 @@ type k8sErrBody struct {
 //
 //nolint:govet
 func parseKubernetesError(responseBytes []byte, statusCode int, err error) error {
-	if err != nil {
-		statusErr := &k8serrors.StatusError{}
-		if errors.As(err, &statusErr) {
-			if statusCode == 0 || (statusErr.ErrStatus.Code > 0 && statusCode != int(statusErr.ErrStatus.Code)) {
-				return NewServerResponseError(statusErr, int(statusErr.ErrStatus.Code))
-			}
-			return NewServerResponseError(statusErr, statusCode)
-		}
-	}
 	if len(responseBytes) > 0 {
-		parsed := k8sErrBody{}
+		var parsed metav1.Status
 		// If we can parse the response body, use the error contained there instead, because it's clearer
 		if e := json.Unmarshal(responseBytes, &parsed); e == nil {
-			err = errors.New(parsed.Message)
-			if statusCode == 0 && parsed.Code > 0 {
-				statusCode = parsed.Code
+			return &k8serrors.StatusError{
```

Comment:
We use the `StatusCode()` method on the various resource Stores to check if a response is a 404 (for upsert, [example](https://github.com/grafana/grafana-app-sdk/blob/main/resource/typedstore.go#L125)), so we'll need to make changes there as well, or wrap both `ServerResponseError` and `StatusError` with an interface that presents the same information.

Reply:
OK, I can take care of that.

Reply:
Updated. I've ended up refactoring error handling in a pretty major way, but on the upside all error handling should now be compatible with k8s api errors package.

Reply:
Looks good, I think the error refactor is cleaner for handling everything!